R/database.R

```diff
@@ -1,30 +1,121 @@
-#' @param tool_name
-#' @param projection
-#' @param seq_type
-#' @param flavour Specify this as either "augmented" or "force_unmatched" or leave out to get the default (original MAF)
+#' @param these_samples_metadata Optional metadata table
+#' @param tool_name Only supports slms-3 currently
+#' @param augmented default: TRUE. Set to FALSE if you instead want the original MAF from each sample for multi-sample patients instead
+#' @param projection Obtain variants projected to this reference (one of grch37 or hg38)
+#' @param flavour Currently this function only supports one flavour option but this feature is meant for eventual compatability with additional variant calling parameters/versions
+#' @param min_read_support Only returns variants with at least this many reads in t_alt_count (for cleaning up augmented MAFs)
+#' @param subset_from_merge Instead of merging individual MAFs, the data will be subset from a pre-merged MAF of samples with the specified seq_type
+
 #'
 #' @return
 #' @export
 #'
 #' @examples
 #' merged_maf_force_unmatched = get_ssm_by_samples(these_sample_ids=c("HTMCP-01-06-00485-01A-01D","14-35472_tumorA","14-35472_tumorB"))
-get_ssm_by_samples = function(these_sample_ids,these_samples_metadata,tool_name="slms-3",projection="grch37",seq_type="genome",flavour=""){
+get_ssm_by_samples = function(these_sample_ids,
+                              these_samples_metadata,
+                              tool_name="slms-3",
+                              projection="grch37",
+                              seq_type="genome",
+                              flavour="clustered",
+                              min_read_support=3,
+                              subset_from_merge=TRUE,
+                              augmented=TRUE){
+  if(!subset_from_merge){
+    message("WARNING: on-the-fly merges can be extremely slow and consume a lot of memory. Use at your own risk. ")
+  }
+  to_exclude = get_excluded_samples(tool_name)
   if(missing(these_samples_metadata)){
-    these_samples_metadata = get_gambl_metadata() %>% dplyr::filter(sample_id %in% these_sample_ids)
+    these_samples_metadata = get_gambl_metadata() %>%
+      dplyr::filter(sample_id %in% these_sample_ids) %>%
+      dplyr::filter(!sample_id %in% to_exclude)
   }else{
-    these_samples_metadata = these_samples_metadata %>% dplyr::filter(sample_id %in% these_sample_ids)
-  }
-  maf_df_list = list()
-  for(this_sample in these_sample_ids){
-    maf_df = get_ssm_by_sample(this_sample,these_samples_metadata,tool_name,projection,seq_type,flavour)
-    maf_df_list[[this_sample]]=maf_df
+    these_samples_metadata = these_samples_metadata %>%
+      dplyr::filter(sample_id %in% these_sample_ids) %>%
+      dplyr::filter(!sample_id %in% to_exclude)
+  }
+  #ensure we only have sample_id that are in the remaining metadata (no excluded/unavailable samples)
+  these_sample_ids = these_sample_ids[which(these_sample_ids %in% these_samples_metadata$sample_id)]
+  if(subset_from_merge){
```

Comment:
Should you add a warning message about legacy maf versions here like you did for `get_ssm_by_sample`? This function could be fixed to pull variants from old merges if we ever want to support that functionality.

Reply:
done. I also left a TODO for whoever tackles this. The implementation in the from_merge approach should actually be a lot easier since it parallels what we used to do